api/src/main/java/org/apache/iceberg/actions/ActionsProvider.java

```diff
@@ -70,4 +70,10 @@ default RewritePositionDeleteFiles rewritePositionDeletes(Table table) {
     throw new UnsupportedOperationException(
         this.getClass().getName() + " does not implement rewritePositionDeletes");
   }
+
+  /** Instantiates an action to analyze tables */
+  default AnalyzeTable analyzeTable(Table table) {
```

Comment:
Question: Have we considered other names like `computeTableStats` or `refreshTableStats` to be a bit more specific? What naming is used in other engines? I'd be curious to hear from everyone.

Reply:
+1, I have a same concern about this generic name

https://github.com/apache/iceberg/pull/10288#discussion_r1593437419

Reply:
I understand there is `ANALYZE` command in Spark but I wonder how to handle partition stats in the future (e.g. like whether they should be computed as part of analyze or separately).

Reply:
[Trino](https://trino.io/docs/current/sql/analyze.html#examples) and [Spark](https://spark.apache.org/docs/3.5.1/sql-ref-syntax-aux-analyze-table.html#analyze-table) seems to be collecting stats specific partition level stats as part of the Analyze command grammar. (Or atleast they dont have a separate command to collect partition stats)
So went with the name, but open to changing it.

Reply:
Maybe by default collect all the stats and have option to compute specific stats like distinct count and partition stats by specifying individually.